lib/Db/MessageMapper.php

```diff
@@ -1220,4 +1220,16 @@ public function resetInReplyTo(): int {
 			);
 		return $update->execute();
 	}
+
+	public function changeMailbox(int $currentId, int $destinationId): int {
+		$qb = $this->db->getQueryBuilder();
+
+		$update = $qb->update($this->tableName)
```

Comment:
this won't work. each IMAP message has a `UID` assigned, but that UID is only unique within the mailbox. If a message is moved from one mailbox to another, a new UID is assigned.

If you only update the mailbox_id, the message UIDs will point to other messages.

It is not trivial to figure out the new UIDs. Therefore I would suggest that we move the messages on IMAP and delete them from the source messages cache. Synchronization will re-add the messages in the trash bin with their new UIDs.

Reply:
OK, just do after that `$this->dbMessageMapper->deleteAll($mailbox);`